syntax = "proto3";
package ParserModel;

message BlobShape {
  repeated int64 dim = 1 [packed = true];
}

message meanParameter {
  repeated float mean = 1;
}

message normParameter {
  repeated float norm = 1;
}
message ModelParameter {
   string name = 1; //consider giving the configParam a name
  repeated InferConfigParameter infer_config = 2; //ID 10 so inferConfigs are printed last.
}

// NOTE
// Update the next available ID when you add a new InferConfigParameter field. 
//
message InferConfigParameter {
   string name = 1;
   string infer_config_type = 2; //config type,such as modelInferEngine(any model), detect, classify or other functional model
   string infer_type = 3; //infer type, such as trt, nb or others
   string model_input_data_format = 4; //GRAY8, BGR24_PLANNER, RGB24_PLANNER
   string model_binary = 5;
   string weight_file = 6;
   uint32 batch_size = 7;
  repeated BlobShape input_shape = 8;
  repeated BlobShape output_shape = 9;
  repeated meanParameter mean_param = 10; 
  repeated normParameter norm_param = 11;
  uint32 max_concurrency = 12;
  uint32 model_position = 13;
}


